`ProjectsService.UploadFile()` is not retryable

When calling `ProjectService.UploadFile()` and get error response at the first request, I expected retry request, but I got following error. 

```
2021/03/28 08:32:20 Post "https://gitlab.com/api/v4/projects/<project id>/uploads": http: ContentLength=254 with Body length 0
```

The cause of this error is rewind request data that do via `Request.body` in go-retryablehttp is not working at retry.
Because this is creating new request with `NewRequest()` with specified `nil` to the 3rd parameter(body).

`ProjectService.UploadFile()`
https://github.com/xanzy/go-gitlab/blob/v0.47.0/projects.go#L1216

`NewRequest()` in go-retryablehttp
https://github.com/hashicorp/go-retryablehttp/blob/v0.6.4/client.go#L226

Thanks for fixing this issue.

I run following test before and after #1115 in my local environment and confirmed that it is fixed.

```go
func TestUploadFile_Retry(t *testing.T) {
    mux, server, client := setup(t)
    defer teardown(server)

    tf, _ := ioutil.TempFile(os.TempDir(), "test")
    defer os.Remove(tf.Name())

    isFirstRequest := true
    mux.HandleFunc("/api/v4/projects/1/uploads", func(w http.ResponseWriter, r *http.Request) {
        if isFirstRequest {
            http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
            isFirstRequest = false
            return
        }
        if false == strings.Contains(r.Header.Get("Content-Type"), "multipart/form-data;") {
                    t.Fatalf("Prokects.UploadFile request content-type %+v want multipart/form-data;", r.Header.Get("Content-Type"))
                }
                if r.ContentLength == -1 {
                    t.Fatalf("Prokects.UploadFile request content-length is -1")
                }
                fmt.Fprint(w, `{
                  "alt": "dk",
                    "url": "/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.md",
                    "markdown": "![dk](/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.png)"
                }`)
    })

    want := &ProjectFile{
        Alt:      "dk",
        URL:      "/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.md",
        Markdown: "![dk](/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.png)",
    }

    file, _, err := client.Projects.UploadFile(1, tf.Name())

    if err != nil {
        t.Fatalf("Prokects.UploadFile returns an error: %v", err)
    }

    if !reflect.DeepEqual(want, file) {
        t.Errorf("Prokects.UploadFile returned %+v, want %+v", file, want)
    }
}
```

before
```
❯ go test -v -run TestUploadFile_Retry
=== RUN   TestUploadFile_Retry
    projects_test.go:406: Prokects.UploadFile returns an error: Post "http://127.0.0.1:49251/api/v4/projects/1/uploads": http: ContentLength=245 with Body length 0
--- FAIL: TestUploadFile_Retry (0.75s)
FAIL
exit status 1
FAIL	github.com/xanzy/go-gitlab	1.223s
```

after
```
❯ go test -v -run TestUploadFile_Retry
=== RUN   TestUploadFile_Retry
--- PASS: TestUploadFile_Retry (0.83s)
PASS
ok  	github.com/xanzy/go-gitlab	1.041s
```

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/xanzy/go-gitlab
 require (
 	github.com/google/go-querystring v1.0.0
 	github.com/hashicorp/go-cleanhttp v0.5.1
-	github.com/hashicorp/go-retryablehttp v0.6.4
+	github.com/hashicorp/go-retryablehttp v0.6.8
 	github.com/stretchr/testify v1.4.0
 	golang.org/x/net v0.0.0-20181108082009-03003ca0c849 // indirect
 	golang.org/x/oauth2 v0.0.0-20181106182150-f42d05182288

--- a/projects.go
+++ b/projects.go
@@ -20,7 +20,6 @@ import (
 	"bytes"
 	"fmt"
 	"io"
-	"io/ioutil"
 	"mime/multipart"
 	"net/http"
 	"os"
@@ -1218,15 +1217,18 @@ func (s *ProjectsService) UploadFile(pid interface{}, file string, options ...Re
 	}
 	w.Close()
 
-	req, err := s.client.NewRequest("", u, nil, options)
+	req, err := s.client.NewRequest(http.MethodPost, u, nil, options)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	req.Body = ioutil.NopCloser(b)
-	req.ContentLength = int64(b.Len())
+	// Set the buffer as the request body.
+	if err = req.SetBody(b); err != nil {
+		return nil, nil, err
+	}
+
+	// Overwrite the default content type.
 	req.Header.Set("Content-Type", w.FormDataContentType())
-	req.Method = http.MethodPost
 
 	uf := &ProjectFile{}
 	resp, err := s.client.Do(req, uf)

--- a/projects_test.go
+++ b/projects_test.go
@@ -367,6 +367,50 @@ func TestUploadFile(t *testing.T) {
 	}
 }
 
+func TestUploadFile_Retry(t *testing.T) {
+	mux, server, client := setup(t)
+	defer teardown(server)
+
+	tf, _ := ioutil.TempFile(os.TempDir(), "test")
+	defer os.Remove(tf.Name())
+
+	isFirstRequest := true
+	mux.HandleFunc("/api/v4/projects/1/uploads", func(w http.ResponseWriter, r *http.Request) {
+		if isFirstRequest {
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			isFirstRequest = false
+			return
+		}
+		if false == strings.Contains(r.Header.Get("Content-Type"), "multipart/form-data;") {
+			t.Fatalf("Prokects.UploadFile request content-type %+v want multipart/form-data;", r.Header.Get("Content-Type"))
+		}
+		if r.ContentLength == -1 {
+			t.Fatalf("Prokects.UploadFile request content-length is -1")
+		}
+		fmt.Fprint(w, `{
+                  "alt": "dk",
+                    "url": "/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.md",
+                    "markdown": "![dk](/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.png)"
+                }`)
+	})
+
+	want := &ProjectFile{
+		Alt:      "dk",
+		URL:      "/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.md",
+		Markdown: "![dk](/uploads/66dbcd21ec5d24ed6ea225176098d52b/dk.png)",
+	}
+
+	file, _, err := client.Projects.UploadFile(1, tf.Name())
+
+	if err != nil {
+		t.Fatalf("Prokects.UploadFile returns an error: %v", err)
+	}
+
+	if !reflect.DeepEqual(want, file) {
+		t.Errorf("Prokects.UploadFile returned %+v, want %+v", file, want)
+	}
+}
+
 func TestListProjectForks(t *testing.T) {
 	mux, server, client := setup(t)
 	defer teardown(server)

